### PR TITLE
Added router

### PR DESCRIPTION
**Pull request by Skullet-on**

### CODE REVIEW

File: src/components/app/App.jsx [hunk 1]
```diff
@@ -3,7 +3,10 @@ import { searchUsers } from "../../helpers/searchUsers";
 import { sortUsers } from "../../helpers/sortUsers";
 import Header from "../header/Header";
 import UsersList from "../usersList/usersList";
+import { BrowserRouter as Router, Switch, Route, Redirect} from "react-router-dom";
+import UserItem from "../userItem/UserItem";
 
+import history from './history';
```
Comment (pavtaras) [inline src/components/app/App.jsx]: no need

File: src/components/header/Header.jsx [hunk 2]
```diff
@@ -9,9 +10,10 @@ class Header extends Component {
 
   handleGetUserLogin(e) {
     e.preventDefault();
+    history.push('/users');
```
Comment (pavtaras) [inline src/components/header/Header.jsx]: it should not be here but in App (in `onSearchUsers` callback)

File: src/components/userItem/UserItem.jsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import React, { Component } from 'react';
+import styles from './userItem.module.css';
+
+class UserItem extends Component {
+  constructor(props) {
```
Comment (pavtaras) [inline src/components/userItem/UserItem.jsx]: no need in constructor here, you can omit constructor:

```js
state = {
      userLogin: "",
      userAvatarUrl: "",
      userName: "",
      createdTime: "",
      userLocation: "",
      userEmail: "",
      userCompany: "",
      userBlog: ""
    }
```

File: src/components/userItem/UserItem.jsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      userName: "",
+      createdTime: "",
+      userLocation: "",
+      userEmail: "",
+      userCompany: "",
+      userBlog: ""
+    }
+  }
+
+  getuser(userLogin) {
```
Comment (pavtaras) [inline src/components/userItem/UserItem.jsx]: `getUser`

File: src/components/userItem/UserItem.jsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    const BASE_URL = "https://api.github.com";
+    return fetch(BASE_URL + "/users/" + userLogin)
```
Comment (pavtaras) [inline src/components/userItem/UserItem.jsx]: create separate function as it was done with `searchUsers`

File: src/components/userItem/UserItem.jsx (hunk 3 above)
Comment (pavtaras) [reply src/components/userItem/UserItem.jsx]: but i would suggest to keep all these props into one 'user' property of state:

```
state = {
  user: {
    userLogin: '...', 
  }
}
```

File: src/components/userItem/UserItem.jsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+      .then(response => response.json())
+  }
+
+  componentDidMount() {
+    this.getuser(this.props.match.params.userlogin)
+      .then(user => {
+        this.setState({
```
Comment (pavtaras) [inline src/components/userItem/UserItem.jsx]: just `this.setState({ user });`

File: src/components/userPage/UserPage.jsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
+import React, { Component } from 'react';
+import styles from './userPage.module.css';
+import { searchUser } from "../../helpers/searchUsers";
+
+class UserPage extends Component {
+  state = {
+    user: []
```
Comment (roman2000s) [inline src/components/userPage/UserPage.jsx]: if we are getting in user objects -we need to rename basic `user:[]` to `user: null`

File: src/components/app/App.jsx [hunk 8]
```diff
@@ -35,10 +42,20 @@ class App extends Component {
           onSearchUsers={this.handleSearchUsers}
           onSortUsers={this.handleSortUsers}
         />
-        <UsersList usersList={this.state.users}/>
+        <Switch>
+          <Route exact path="/users">
+            <UsersList usersList={this.state.users} />
+          </Route>
```
Comment (pavtaras) [inline src/components/app/App.jsx]: lets also write code with render (for consistency):

```jsx
<Route 
  exact
  path="/users"
  render={() => <UsersList usersList={this.state.users} />}
/>
```

File: src/components/app/App.jsx [hunk 9]
```diff
@@ -35,10 +42,20 @@ class App extends Component {
...
+          <Route path="/users/:userlogin" render={(params) => <UserPage userLogin={params.match.params.userlogin} />} />
+          <Route>
+            <Redirect to={{
+              pathname: "/users"
+            }} />
+          </Route>
```
Comment (pavtaras) [inline src/components/app/App.jsx]: here too, use render pls

File: src/components/header/Header.jsx [hunk 10]
```diff
@@ -11,7 +11,7 @@ class Header extends Component {
     e.preventDefault();
     this.state.userlogin ? this.props.onSearchUsers(this.state.userlogin) : alert("Login should not be empty");
   }
-
+
```
Comment (pavtaras) [inline src/components/header/Header.jsx]: remove it

**Pull request merged**